src/main.cpp

```diff
@@ -258,25 +255,82 @@ namespace rhost {
             log::logf(log_verbosity::minimal, ok ? "Workspace loaded successfully.\n" : "Failed to load workspace.\n");
         }
 
-#ifdef _WIN32
         UINT_PTR timer = SetTimer(NULL, IDT_RESET_TIMER, 5000, [](HWND hWnd, UINT msg, UINT_PTR idEVent, DWORD dwTime) {
             rhost::host::do_r_callback(false);
         });
 
         SCOPE_WARDEN(destroy_timer, {
             KillTimer(NULL, timer);
         });
-#endif
 
         run_Rmainloop();
 
-        Rf_endEmbeddedR(0);
         return EXIT_SUCCESS;
     }
 
+#else // POSIX
+    int run_r_posix(command_line_args& args) {
+        init_log(args.name, args.log_dir, args.log_level, args.suppress_ui);
+        transport::initialize();
+        
+        R_running_as_main_program = 1;
+        
+        ptr_R_ShowMessage = rhost::host::ShowMessage;
+
+         char *argv[] = {"Microsoft.R.Host", "--interactive"};
+        int argc = sizeof(argv)/sizeof(argv[0]);
+        int res = Rf_initialize_R(argc, argv);
+
+        structRstart rp = {};
+        R_DefParams(&rp);
+        rp.R_Quiet = R_FALSE;
+        rp.R_Interactive = args.is_interactive ? R_TRUE : R_FALSE;
+        rp.RestoreAction = SA_NORESTORE;
+        rp.SaveAction = SA_NOSAVE;
+
+        rhost::host::initialize(rp, args.rdata, args.idle_timeout);
+
+        R_set_command_line_arguments(args.argc, args.argv.data());
+        R_common_command_line(&args.argc, args.argv.data(), &rp);
+        R_SetParams(&rp);
+        
+        // This is a exported static library member Rf_initialize_R sets it to TRUE
+        R_Interactive = args.is_interactive ? R_TRUE : R_FALSE;
+        R_Consolefile = nullptr;
+        R_Outputfile = nullptr;
+
+        DllInfo *dll = R_getEmbeddingDllInfo();
+        rhost::r_util::init(dll);
+        rhost::grdevices::xaml::init(dll);
+        rhost::grdevices::ide::init(dll);
+        rhost::exports::register_all(dll);
+
+        rhost::host::setCallbacksPOSIX();
+
+        if (!args.rdata.empty()) {
+            std::string s = args.rdata.string();
+            log::logf(log_verbosity::minimal, "Loading workspace from file %s\n", s.c_str());
+
+            bool ok = r_top_level_exec([&] {
+                R_RestoreGlobalEnvFromFile(s.c_str(), R_FALSE);
+            });
+
+            log::logf(log_verbosity::minimal, ok ? "Workspace loaded successfully.\n" : "Failed to load workspace.\n");
+        }
+
+        Rf_mainloop();
+
```

Comment:
MRO does exist on Linux. We should still be suggesting it. https://mran.microsoft.com/download/